x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/execution/ExecutionService.java

```diff
@@ -278,8 +279,10 @@ public WatchRecord execute(WatchExecutionContext ctx) {
                         if (resp.isExists() == false) {
                             throw new ResourceNotFoundException("watch [{}] does not exist", watchId);
                         }
-                        return parser.parseWithSecrets(watchId, true, resp.getSourceAsBytesRef(), ctx.executionTime(), XContentType.JSON,
-                            resp.getSeqNo(), resp.getPrimaryTerm());
+                        Watch watch =  parser.parseWithSecrets(watchId, true, resp.getSourceAsBytesRef(), ctx.executionTime(),
+                            XContentType.JSON, resp.getSeqNo(), resp.getPrimaryTerm());
+                        watch.version(resp.getVersion());
```

Comment:
I might be missing something here... but won't the WatchParser throw an exception if `_version` is found in the response ?

The old code used to do this:
```
          } else if (WatchField.VERSION.match(currentFieldName, parser.getDeprecationHandler())) {
                version = parser.longValue();
            } ....
           } else {
                throw new ElasticsearchParseException("could not parse watch [{}]. unexpected field [{}]", id, currentFieldName);
            }
```
The new code doesn't have this branch
```
         } else if (WatchField.VERSION.match(currentFieldName, parser.getDeprecationHandler())) 
```
so it will just fall to the `else` to the Exception ?

Reply:
Good question. It looks like the version was never persisted. In `Watch.java` the version field was never serialized as xcontent (it just existed in that file as a transient field). I'm not sure why this version parsing code existed.

Reply:
I could still be missing something here ... but isn't this the field `_version` ? Which if you save a watch (in 6.6) and look at it in the index, it is persisted.

Reply:
I think the _version parse logic was bogus. 

It was never persisted:
https://github.com/elastic/elasticsearch/commit/c7ef318a503881853d5075d13848d48afd4b6c5c#diff-ce906733bb0a3448a269bd2e1897af23L40

and when really needed was provided differently:
https://github.com/elastic/elasticsearch/commit/c7ef318a503881853d5075d13848d48afd4b6c5c#diff-61c4432a1abf7b920802c0251960bb52L343

Reply:
(these are links to the commit that removed _version in favour of seqno and primary term: c7ef318a503881853d5075d13848d48afd4b6c5c)

Reply:
we discussed in person, and now see that parsing logic was extraneous code.